Instant Search: Remove `debug` from production build

sgomes opened the pull request:
The `debug` npm module is currently only used by `photon`. Since that debug information is not particularly useful in production, this PR replaces it with a dummy implementation in the production build, using `webpack`'s built-in `NormalModuleReplacementPlugin`.

This should remove around 6KB (2KB gzipped) from the main Instant Search bundle.

#### Changes proposed in this Pull Request:
* Instant Search: remove `debug` from production bundle

#### Jetpack product discussion
None.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
Some simple smoke-testing to ensure that Instant Search still works correctly should be enough, particularly if you're able to bring up search results with images.
I tested this in my wpcom sandbox environment, but I couldn't get any images to be shown on the test site I used.

#### Proposed changelog entry for your changes:
* No need for a changelog entry; this should be a seamless under-the-hood change.

sgomes committed:
Remove `debug` from prod Instant Search build

jsnmoon reviewed (dismissed):
Photonized images still load as expected. Confirmed size reduction from 46.05 kB gzipped to 43.95 kB gzipped 👍.

sgomes committed:
Add changelog entry

sgomes commented:
Even though this PR preceded the changelogger requirement, it was still being blocked by the changelogger merge check.

I rebased and added a changelogger entry, but unfortunately this reset the review flag, which means I still can't merge. Would you mind re-approving, @jsnmoon? Thank you!

sgomes commented:
Thanks again, @jsnmoon! 👍

sgomes commented:
Changeset ID 222898
